src/arch/host/lib/ams.c

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2023 Intel Corporation. All rights reserved.
+//
+// Author: Krzysztof Frydryk <krzysztofx.frydryk@intel.com>
+
+/**
+ * \file
+ * \brief Xtensa Asynchronous Messaging Service implementation file
+ * \authors Krzysztof Frydryk <krzysztofx.frydryk@intel.com>
+ */
+
+#include <sof/lib/cpu.h>
+#include <sof/lib/ams.h>
+#include <xtos-structs.h>
+
+static struct async_message_service *host_ams;
+
+struct async_message_service **arch_ams_get(void)
+{
+	return &host_ams;
+}
```

Comment:
Every time I compared the generated binary with versus without `inline`, I never found any difference. I think most compilers ignore `inline` now.

`static inline` can be useful in .h files to avoid "unused function" warnings but I don't think `inline` makes any difference in .c files. You can use it as a hint... for the readers :-)

Reply:
My bad, did not notice that it's a .c file not a header.